modules/core/src/main/scala/org/scalasteward/core/buildtool/sbt/SbtAlg.scala

```diff
@@ -55,8 +55,17 @@ final class SbtAlg[F[_]](config: Config)(implicit
     for {
       buildRootDir <- workspaceAlg.buildRootDir(buildRoot)
       maybeSbtVersion <- getSbtVersion(buildRootDir)
-      lines <- addStewardPluginTemporarily(buildRootDir, maybeSbtVersion).surround {
-        val commands = Nel.of(crossStewardDependencies, reloadPlugins, stewardDependencies)
+      metaBuilds = Math.max(
+        LazyList
+          .iterate(buildRootDir / project)(_ / project)
+          .takeWhile(p => p.exists && p.isDirectory)
+          .size,
+        1 // There is always at least one meta build, even if there is no project folder
```

Comment:
I'm wondering if we should drop this `Math.max` call. If there is no `project` folder, there aren't any `*.sbt` files either which could contain dependencies (plugins) that Scala Steward could update. And if there is nothing to update, why loading the meta-build and calling `stewardDependencies`? :thinking:

Reply:
In theory yes. However removing `Math.max` will break many many tests because of lines like
https://github.com/scala-steward-org/scala-steward/blob/a6ff326cb8c7d423ccbd5a656a771d1ef9c634a4/modules/core/src/test/scala/org/scalasteward/core/buildtool/sbt/SbtAlgTest.scala#L29
https://github.com/scala-steward-org/scala-steward/blob/a6ff326cb8c7d423ccbd5a656a771d1ef9c634a4/modules/core/src/test/scala/org/scalasteward/core/buildtool/sbt/SbtAlgTest.scala#L43

which assume scala-steward always checks/updates the `project` folder. And I would keep all those tests how they are currently.
Also in real life I never saw a sbt project without a `project` folder (sbt even creates the `project` folder when running in an empty folder).

I would just keep it.

Reply:
> And I would keep all those tests how they are currently.

We can keep them if we use `FileAlg#isDirectory` for checking the `project` folders as I've done in  https://github.com/scala-steward-org/scala-steward/pull/2995/commits/1a97fc56c00bdaa9b468d4b6ae52ed87cd351489. Another benefit of using `FileAlg` is that the tests are now sensitive to the actual number of `project` folders.

Reply:
Great! Thanks!